cmd/sonrd/utils/keychain.go

```diff
@@ -0,0 +1,123 @@
+package utils
+
+import (
+	"bytes"
+	"encoding/gob"
+
+	"github.com/pkg/errors"
+	"github.com/sonr-io/keyring"
+	"github.com/sonr-io/sonr/pkg/crypto/mpc"
+	"github.com/sonr-io/sonr/pkg/fs"
+	mt "github.com/sonr-io/sonr/third_party/types/motor"
+)
+
+const K_SERVICE_NAME = "sonr-dev"
+
+type UserAuth struct {
+	Password  string
+	AesDSCKey []byte
+	AesPSKKey []byte
+}
+
+func (i UserAuth) Validate() bool {
+	if len(i.Password) < 12 {
+		return false
+	}
+	if i.AesDSCKey == nil {
+		return false
+	}
+	if len(i.AesDSCKey) < 32 {
+		return false
+	}
+	return true
+}
+
+func (i UserAuth) GenAccountCreateRequest() (*mt.CreateAccountRequest, error) {
+	if i.Validate() {
+		return &mt.CreateAccountRequest{
+			Password:  i.Password,
```

Comment:
Assuming here this is the password in flight. Please let me know if this assumption is incorrect.

Reply:
Can you clarify in flight?

Reply:
sent across the wire for authentication purposes, ie a web login form